libs/utils/analysis/latency_analysis.py

```diff
@@ -331,6 +344,123 @@ def plotLatencyBands(self, task, axes=None):
+        if wkp_df is None:
+            return None
+        self._log.info('Found: %5d activations for [%s]',
+                       len(wkp_df), td.label)
+
+        # Disregard data above two time the specified threshold
+        y_max = (2 * threshold_ms) / 1000.
+        len_tot = len(wkp_df)
+        wkp_df = wkp_df[wkp_df.activation_interval <= y_max]
+        len_plt = len(wkp_df)
+        if len_plt < len_tot:
+            len_dif = len_tot - len_plt
+            len_pct = float(len_dif) / len_tot
+            self._log.warning('Discarding {} activation intervals (above 2 x threshold_ms, '
+                              '{:.1f}% of the overall activations)'\
+                              .format(len_dif, len_pct))
+        ymax = 1.1 * wkp_df.activation_interval.max()
+
+        # Setup plots
+        gs = gridspec.GridSpec(2, 2, height_ratios=[2,1], width_ratios=[1,1])
+        plt.figure(figsize=(16, 8))
+
+        plot_title = "Activaton intervals [{}] (at wakeup time)".format(td.label)
+        if tag:
+            plot_title = "{} [{}]".format(plot_title, tag)
+        plot_title = "{}, threshold @ {} [ms]".format(plot_title, threshold_ms)
+
+        # Activations intervals over time
+        axes = plt.subplot(gs[0,0:2])
+        axes.set_title(plot_title)
+        wkp_df.plot(style='g+', logy=False, ax=axes)
+
+        axes.axhline(threshold_ms / 1000., linestyle='--', color='g')
+        self._trace.analysis.status.plotOverutilized(axes)
+        axes.legend(loc='lower center', ncol=2)
+
+        # Cumulative distribution of all running times
+        axes = plt.subplot(gs[1,0])
+        df = pd.DataFrame(sorted(wkp_df.activation_interval),
+                          columns=['activation_interval'])
+        df.activation_interval.plot(ax=axes, logy=True, legend=False,
+                                    title='Activation intervals cumulative '
+                                          '(red: {} [ms] threshold)'\
+                                          .format(threshold_ms))
+        axes.axhline(y=threshold_ms / 1000., linewidth=2,
+                     color='r', linestyle='--')
+
+        # Histogram of all running times
+        axes = plt.subplot(gs[1,1])
+        wkp_df.plot(kind='hist', bins=64, ax=axes,
+                        xlim=(0,ymax), legend=False,
+                        title='Activation intervals histogram (64 bins, {} [ms] green threshold)'\
+                        .format(threshold_ms));
+        axes.axvspan(0, threshold_ms / 1000., facecolor='g', alpha=0.5);
+
+        # Save generated plots into datadir
+        task_name = re.sub('[\ :/]', '_', td.label)
+        figname = '{}/{}task_activations_{}_{}.png'\
+                  .format(self._trace.plots_dir, self._trace.plots_prefix,
+                          td.pid, task_name)
+        pl.savefig(figname, bbox_inches='tight')
+
+        # Return statistics
+        return wkp_df.describe(percentiles=[0.95, 0.99])
```

Comment:
I don't have any matplotllib fu so can't give much input on this code but the output looks great :+1: :stuck_out_tongue: